### PR TITLE
Initial smoke-test framework for docker testimages

### LINKED ISSUE #8938: [CI] Implement smoke-test for container images created by Neighbors
**Description**

Each container image which is created by Neighbors has to be checked by a smoke-test to ensure that the image is still behaving / working as expected. The smoke test should just apply a simple check of the installed tools to ensure we have no incompatibly or version conflicts.

AC:
- [ ] Define a mechanism how a smoke test (e.g. a bash-script or similar) could be executed for a new created container image
- [ ] Output of failing smoke-tests are collected and reported at the end of the executed build process.
- [ ] A failing smoke test should not interrupt the build loop
- [ ] If the build process includes a failing smoke-test, the build job has to be marked as failed 

**Reasons**

Ensure created / updated container images are working as expected and not including obvious issues.

**Acceptance Criteria**

If a smoke test is defined for a container image, it has to be executed after a new container image was created.

**Attachments**

### PR DESCRIPTION
**Pull request by Ressetkk**
/kind feature
/area ci

Added small framework that allows to define small test script that allows doing some work around the Docker context. I've added small tests around basic functionalities and binary presence. Script results outputs are collected to the ARTIFACTS directory and if any error occurs, the build loop is interrupted, returning non-zero exit code. The functionality docu is provided for anyone who wants to add another smoke test for new or existing image.

Closes #8938

### CODE REVIEW

File: images/README.md [hunk 1]
```diff
@@ -10,3 +10,22 @@ To add additional applications into the images, open a PR with changes. Follow t
 * Always build from a source to ensure compiler vulnerabilities do not affect the resulting binary
 * Link the binary to a specific version so that it's easier to update when necessary 
 * Build binaries in a separate stage, then copy the resulting binary into the final image to ensure images are small and contain the least number of layers
+
+## Writing image tests
```
Comment (IwonaLanger) [inline images/README.md:14]:
```suggestion
## Write image tests
```

File: images/README.md [hunk 2]
```diff
@@ -10,3 +10,22 @@ To add additional applications into the images, open a PR with changes. Follow t
...
+
+to write simple smoke tests with your image, add an **executable** file called `test.sh`.
```
Comment (IwonaLanger) [inline images/README.md:16]:
```suggestion
To write simple smoke tests with your image, add an executable file called `test.sh`.
```

File: images/README.md [hunk 3]
```diff
@@ -10,3 +10,22 @@ To add additional applications into the images, open a PR with changes. Follow t
...
+The scripts should contain all steps that perform basic or advanced test operations against the image. You are allowed to use all binaries available in [E2E DinD K3d image](./e2e-dind-k3d) to test built image.
```
Comment (IwonaLanger) [inline images/README.md:17]:
```suggestion
The scripts should contain all steps that perform basic or advanced test operations against the image. You can use all binaries available in [E2E DinD K3d image](./e2e-dind-k3d) to test the built image.
```

File: images/README.md [hunk 4]
```diff
@@ -10,3 +10,22 @@ To add additional applications into the images, open a PR with changes. Follow t
...
+Test script **must** exit with non-zero number, if any of the steps failed.
```
Comment (IwonaLanger) [inline images/README.md:18]:
```suggestion
The test script must exit with a non-zero number if any steps have failed.
```

File: images/README.md [hunk 5]
```diff
@@ -10,3 +10,22 @@ To add additional applications into the images, open a PR with changes. Follow t
...
+
+By default, current context of a test script will always be Docker build context. Image name is passed as a variable `IMG`.
+
+### Example
+
+The example below showcases the example definition of test.sh script.
```
Comment (IwonaLanger) [inline images/README.md:24]:
```suggestion
The example below showcases the example definition of the `test.sh` script.
```

File: images/README.md (hunk 2 above)
Comment (Ressetkk) [reply images/README.md:16]: I don't understand. I've purposely set the `executable` word in bold to highlight that this script must executable.

File: images/README.md (hunk 1 above)
Comment (Ressetkk) [reply images/README.md:14]: It's continuous work. Should it really be written in simple than continuous?

File: images/README.md (hunk 2 above)
Comment (IwonaLanger) [reply images/README.md:16]: Please, see what we use bold font for in the [guidelines](https://github.com/kyma-project/community/blob/main/docs/guidelines/content-guidelines/03-formatting.md#use-bold-font-for-these-items).

File: images/README.md (hunk 1 above)
Comment (IwonaLanger) [reply images/README.md:14]: Please read the second bullet point [here](https://github.com/kyma-project/community/blob/main/docs/guidelines/content-guidelines/03-formatting.md#headings).

Comment (Ressetkk) [reply images/README.md:14]: Meh.

**Review by IwonaLanger — APPROVED** [review]

**Review by halamix2 — APPROVED** [review]
/pony

**Pull request merged**